### PR TITLE
fix: update UMP transfers tracking status

### PR DESCRIPTION
**Pull request by Victor-Salomon**
This PR updates the tracking status for UMP transfers (ex: DOT from AH to Relaychain or DOT from Biforst to Hydration (two-step transfer of non-native token)). 

It also contains some code improvements including: outsourcing helpers, function renaming, and type updates.

Paths tested:
- DOT from AH to Relaychain
- DOT from Relaychain to AH
- DOT from Biforst to Hydration
- BNC from Biforst to Hydration

### CODE REVIEW

File: app/src/models/subscan.ts [hunk 1]
```diff
@@ -1,14 +1,22 @@
 import { TransferStatus } from '@snowbridge/api/dist/history'
 
-export type SubscanTransferResponse = {
+export type Status = 'pending' | 'relayed' | 'success' | 'failed'
+
+type Metadata = {
+  send_at?: number
```
Comment (NunoAlexandre) [inline app/src/models/subscan.ts:6]: do you know what `send_at` is? is it a block number?

File: app/src/utils/subscan.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import { subscan, environment } from '@snowbridge/api'
 import { TransferStatus } from '@snowbridge/api/dist/history'
-import { SubscanTransferResponse, FromParachainTrackingResult } from '@/models/subscan'
+import { SubscanTransferResponse, FromParachainTrackingResult, Status } from '@/models/subscan'
 import { OngoingTransferWithDirection } from '@/models/transfer'
 
-export const trackFromParachainTx = async (
+export const trackWithinPolkadotTx = async (
```
Comment (NunoAlexandre) [inline app/src/utils/subscan.ts:6]: We could shorten it and make it more specific:

```suggestion
export const trackXcm = async (
```

File: app/src/utils/transferTracking.ts [hunk 3]
```diff
@@ -20,15 +20,15 @@ export const trackTransfers = async (
...
+    console.log('From AH To Ethereum transfers:', ahToEthereumTx.length)
     transfers.push(...ahToEthereumTx)
   }
 
   // Keep this until we can test & check tracking process for from Para to AH transfers
   if (ongoingTransfers.withinPolkadot.length) {
-    const parachainToAhTx = await trackFromParachainTx(env, ongoingTransfers.withinPolkadot)
-    console.log('From Parachain To AH transfer:', parachainToAhTx.length)
-    transfers.push(...parachainToAhTx)
+    const withinPolkadotTx = await trackWithinPolkadotTx(env, ongoingTransfers.withinPolkadot)
```
Comment (NunoAlexandre) [inline app/src/utils/transferTracking.ts:29]:
```suggestion
    const xcmTx = await trackWithinPolkadotTx(env, ongoingTransfers.withinPolkadot)
```

**Review by NunoAlexandre — APPROVED** [review]
Looks good to me, just a question and a small comment 🙏

File: app/src/models/subscan.ts (hunk 1 above)
Comment (Victor-Salomon) [reply app/src/models/subscan.ts:6]: Unfortunately, subscan do not provide a type for their response, or I have not found any. 
So I recreated it, this is why it uses `'_'` and not js camelCase.

Metadata are only provided for a transfer to ETH. `send_at` returns a timestamp in milliseconds so a number.

**Review by NunoAlexandre — APPROVED** [review]
🛺  let's go!

**Pull request merged**